components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/GetCookieFunctionImpl.java

```diff
@@ -103,7 +106,15 @@ public String getCookieValue(JsServletRequest request, Object... params) {
                         try {
                             String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext()
                                     .getTenantDomain();
-                            boolean isValid = IdentityUtil.validateSignatureFromTenant(valueString, signature, tenantDomain);
+                            HTTPFunctionsServiceHolder.getInstance().getIdentityKeyStoreGenerator()
+                                    .generateContextKeyStore(tenantDomain, KEY_STORE_CONTEXT);
```

Comment:
Do we need the method here? because if the new keystore has not been generated while the cookie is being set the signature validation is guaranteed to fail if we generate the keystore here.

Reply:
When getCookie called first before setCookie, then this keystore won't be there in the system and throw exception thats why I add create method also.

Reply:
fixed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/198/commits/2b04eef2d59ee7c8cc76f4882c81e82efa0e7564